frontend/src/components/pages/CampCreation/index.tsx

```diff
@@ -1,58 +1,95 @@
+  const [campDetailsDummyThree, setCampDetailsDummyThree] = useState("");
+  const [scheduleSessionsDummyOne, setScheduleSessionsDummyOne] = useState(
+    false,
+  );
+  const [scheduleSessionsDummyTwo, setScheduleSessionsDummyTwo] = useState(
+    false,
+  );
+  const [registrationFormDummyOne, setRegistrationFormDummyOne] = useState(
+    false,
+  );
+  const [registrationFormDummyTwo, setRegistrationFormDummyTwo] = useState(
+    false,
+  );
+
+  // State to determine whether or not all required fields have been filled out.
+  // NOTE: This will depend on what type of state a page requires, i.e. determining
+  // if a checkbox is checked is different than determining if an input field is filled.
+
+  const isCampDetailsFilled =
+    campDetailsDummyOne && campDetailsDummyTwo && campDetailsDummyThree !== "";
+  const isScheduleSessionsFilled =
+    scheduleSessionsDummyOne && scheduleSessionsDummyTwo;
+  const isRegistrationFormFilled =
+    registrationFormDummyOne && registrationFormDummyTwo;
 
-const CampCreationPage = (): JSX.Element => {
   const [currentPage, setCurrentPage] = useState<CampCreationPages>(
-    CampCreationPages.CampCreationDetailsPage,
+    CampCreationPages.CampDetailsPage,
   );
 
-  const getCampCreationStepComponent = (nextPage: CampCreationPages) => {
-    // will also need logic blocking pages if current page not complete
-    switch (nextPage) {
-      case CampCreationPages.CampCreationDetailsPage:
-        return <CampDetails />;
-      case CampCreationPages.ScheduleSessionsPage:
-        return <ScheduleSessions />;
-      case CampCreationPages.RegistrationFormPage:
-        return <RegistrationForm />;
-      default:
-        throw new Error("never reached");
-    }
-  };
-
   return (
-    <Box>
-      <Button
-        onClick={() =>
-          setCurrentPage(CampCreationPages.CampCreationDetailsPage)
-        }
-      >
-        Camp Details
-      </Button>
-      <Button
-        onClick={() => setCurrentPage(CampCreationPages.ScheduleSessionsPage)}
-      >
-        Schedule Sessions
-      </Button>
-      <Button
-        onClick={() => setCurrentPage(CampCreationPages.RegistrationFormPage)}
-      >
-        Registration Form
-      </Button>
-
-      <Box my="50px" mx="228px">
-        {getCampCreationStepComponent(currentPage)}
+    <>
+      <Stepper
+        currentPage={currentPage}
+        setCurrentPage={setCurrentPage}
+        isCampDetailsFilled={isCampDetailsFilled}
+        isScheduleSessionsFilled={isScheduleSessionsFilled}
+        isRegistrationFormFilled={isRegistrationFormFilled}
+      />
+      <Box m={20}>
+        {currentPage === CampCreationPages.CampDetailsPage && (
+          <CampDetails
+            campDetailsDummyOne={campDetailsDummyOne}
+            campDetailsDummyTwo={campDetailsDummyTwo}
+            campDetailsDummyThree={campDetailsDummyThree}
+            toggleCampDetailsDummyOne={() =>
```

Comment:
It seems redundant but we probably need to pass in both getters and setters for each piece of state as props... since these are controlled components, something like a checkbox would need both the getter and setter.

Reply:
I think it's fine given that we don't have something like redux set up. If it gets too ugly, we can do what I did for edit camper info where we would defined a type for each page for setters and gettters and then pass in that const instead of listing all the props.